File: dodam-core/src/main/java/b1nd/dodamcore/wakeupsong/domain/exception/WakeupSongExceptionCode.java
package b1nd.dodamcore.wakeupsong.domain.exception;

import b1nd.dodamcore.common.exception.ExceptionCode;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@RequiredArgsConstructor
public enum WakeupSongExceptionCode implements ExceptionCode {

    ALREADY_APPLIED(HttpStatus.LOCKED, "이미 이번주에 기상송을 신청함"),
    NOT_FOUND(HttpStatus.NOT_FOUND, "없는 기상송"),
    NOT_APPLICANT(HttpStatus.FORBIDDEN, "신청자가 아님"),
    URL_MALFORMED(HttpStatus.BAD_REQUEST, "잘못된 유튜브 URL 형식");

    private final HttpStatus status;
    private final String message;

    @Override
    public HttpStatus getHttpStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }

}

File: services/service-wakeup-song/src/main/kotlin/com/b1nd/dodamdodam/wakeupsong/domain/wakeupsong/exception/WakeupSongExceptionCode.kt
package com.b1nd.dodamdodam.wakeupsong.domain.wakeupsong.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class WakeupSongExceptionCode(
    override val status: HttpStatus,
    override val message: String
) : ExceptionCode {
    WAKEUP_SONG_NOT_FOUND(HttpStatus.NOT_FOUND, "기상송을 찾을 수 없어요."),
    WAKEUP_SONG_NOT_OWNER(HttpStatus.FORBIDDEN, "본인의 기상송만 삭제할 수 있어요."),
    YOUTUBE_VIDEO_NOT_FOUND(HttpStatus.NOT_FOUND, "유튜브 영상을 찾을 수 없어요."),
    MELON_CHART_FETCH_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "멜론 차트를 가져올 수 없어요."),
    AUDIO_DOWNLOAD_FAILED(HttpStatus.INTERNAL_SERVER_ERROR, "오디오 다운로드에 실패했어요."),
    ;
}
